src/index.js

```diff
@@ -574,9 +574,20 @@ class EliteDangerousJournalServer {
   getFileUpdate(filepath) {
     const baseName = path.basename(filepath).toLowerCase();
     if (this.ancillaryFiles[baseName]) {
-      const parsedEvent = JSON.parse(fs.readFileSync(filepath, 'utf-8'));
-      if (parsedEvent) {
-        this.broadcast(Object.assign({}, parsedEvent, this.ancillaryFiles[baseName]));
+      try {
+        const parsedEvent = JSON.parse(fs.readFileSync(filepath, 'utf-8'));
+        if (parsedEvent) {
+          this.broadcast(Object.assign({}, parsedEvent, this.ancillaryFiles[baseName]));
+        }
+      }
+      catch (err) {
+        /* If the file is in the middle of being flushed to disk, we can
+         * get 'SyntaxError: Unexpected end of JSON input' in JSON.parse().
+         */
+        if (err.name != 'SyntaxError') {
+          throw err;
+        }
+        console.log(`${chalk.red('Ignoring error:')} ${chalk.magenta(err)}`);
```

Comment:
It's totally out of scope for this PR, but do you think it would be useful to have a `DEBUG` environment variable that would let you enable/disable this kind of logging for development/testing purposes?

Reply:
I have a few other questions I'd love to get your opinion on. They are outside the scope of this PR, but I'd really appreciate your thoughts on them.

1) I know the Journal's JSON Lines implementation is a little different than how the `status.json` file is handled, but do you think we should add a similar guard around the reading of the Journal? I never encountered this kind of error with it while testing and playing, but that could have just been blind luck.

2) Is the current read interval quick enough to pick up the changes without a noticeable delay when this error is encountered or should we consider adding a short retry interval when this happens? You are far more familiar with it, so if you're happy with the current performance then I think we're probably good.

Reply:
Hey Eric

Sorry for the delay on my side, I got a bit swamped with other stuff. Let me gather my thoughts on your questions and I'll get back to you soon.